### ISSUE TITLE
Fit model more than one time

### ISSUE THREAD

**Issue by Fheanor** [fchollet/keras#3094]
Hey !

First, I am really a super beginner in Deep Learning so I might not be to use correct term, I am sorry.

I built a model with 3 layers that work with pictures as datas and int as labels. It is a supervised model.
To load the datas, I build a very larg matrix and each line is a picture.
Then I use the function 'fit' to train the model.

Example : 

```
#Loading some datas in X_train
#Y_train is a matrix of my labels

#Loading part ...

#-----------------------------------

Y_train = np_utils.to_categorical(y_train,nb_classes)

model = Sequential()
model.add(Dense(1, input_shape=(12288,)))
model.add(Activation('relu'))
model.add(Dropout(0.2))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dropout(0.2))
model.add(Dense(34074)) 
model.add(Activation('softmax'))

model.compile(loss='categorical_crossentropy',
              optimizer=RMSprop(),
              metrics=['accuracy'])

history = model.fit(X_train, Y_train,
                    batch_size=batch_size, nb_epoch=nb_epoch,
                    verbose=1)
```

However, when my matrix is too big, I have a memory Error problem when I use "np_utils.to_categorical(..).
So I would like to split my matrix in 2 matrix and then it might work. But I will need to call the "fit" function two times.
 Does anyone have any idea if it could work or not ?

Thanks a lot and have a good day

**Comment by Fheanor** [comment]
For information :

Each image is associated with a integer.
I have at this time  34074 images.

So because I need 34074 different labels,` Y_train = np_utils.to_categorical(y_train,nb_classes)` try to create a 34074*34074 matrix, that it seems it way too big.

I saw that that instead of using "fit", you can use "train_on_bashes" function.
But the problem is that I don't know how to use it in this case. My last layer need to have a 34074 dense because of the 34074 differents labels.

I tried something like to use "train_on_bashes" function with 500 datas image at the same time in a loop. But my "Y_train" matrix is 500*500 so it can't work because my output layer need a 34074*34074 matrix...

It is a bit confused to explain because english is not my first langage, so if you need any more detail, I would be happy to answer you.
And I am sorry if I say really stupid things, but I just start DeepLearning.

Best regards.

**Comment by AdityaGudimella** [comment]
Why would you have 34074 labels? What are you trying to achieve by using the NNet? Are you trying to classify the pictures into categories? Generally, the number of labels is much lesser than the number of data points (images) that you have. Are you sure there are 34074 labels?

**Comment by Fheanor** [comment]
In fact, I work in gesture recognitions. And at this time, our aim is to build a 3D hand that looks like the same that the user is doing in front of the camera.
To do this, we give to the model lot of images of different positions of a hand and I want each image to be associated with a label.
This label is a number that allow me to find the parameters needed to build a 3D hand.
In my example, I work with 34074 different hand positions so I need to be able to generate 34074 3D hands, so I need 34074 labels...
So there is one category for each image.

**Comment by jmhessel** [comment]
I think that your learning task is not well formulated, deep learning aside.

If you have as many discrete labels as you have training examples, it's going to be very difficult for any model to generalize to new examples. The core of the problem is that you only have a single example for each class. There are proper ways to do "one shot" learning, but treating each image as it's own class is probably not one of them.

In my view, you should potentially discard the supervised learning framework you have in mind and look into something unsupervised.

Have you considered using an autoencoder? In short, an auto-encoder is a model that predicts its own inputs, but through an informational bottleneck. For example, if you input is 12288 dimensional, you could train a model that maps from 12288 dimensions to 200 dimensions, and then back to the 12288 dimensions. Your inputs and outputs could be the same. This will force your network to learn a 200-D compression of your input data. You can then take your new training examples, put them into your learned encoder, and find nearest-neighbors in the 200-D space.

**Comment by Fheanor** [comment]
Thank you for your short explanation, the classification solution cannot work in my case I think.
I never considered to use auto encoder but I will check how it work !
